lib/diff.go

```diff
@@ -232,9 +235,44 @@ func (r *DatasetRequests) Diff(p *DiffParams, res *DiffResponse) (err error) {
 		return err
 	}
 
-	res.Stat = &deepdiff.Stats{}
-	res.A = leftData
-	res.B = rightData
-	res.Diff, err = deepdiff.Diff(leftData, rightData, deepdiff.OptionSetStats(res.Stat))
+	dd := deepdiff.New()
+	res.Diff, res.Stat, err = dd.StatDiff(ctx, leftData, rightData)
 	return err
 }
+
+func schemaDiff(ctx context.Context, left, right *component.BodyComponent) ([]*Delta, *DiffStat, error) {
+	dd := deepdiff.New()
+	if left.Format == ".csv" && right.Format == ".csv" {
+		left, err := terribleHackToGetHeaderRow(left.InferredSchema)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		right, err := terribleHackToGetHeaderRow(right.InferredSchema)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		return dd.StatDiff(ctx, left, right)
+	}
+	return dd.StatDiff(ctx, left.InferredSchema, right.InferredSchema)
+}
+
+// TODO - holy shit dis so bad. fix
```

Comment:
Can you elaborate why this is needed, what the problem is that it's temporarily working around, and what needs to be done instead to fix it?